client/source/js/modules/model/cost-coverage-ctrl.js

```diff
@@ -385,19 +387,18 @@ define(['./module', 'underscore'], function (module, _) {
       var model = getPlotModel();
       model.coparams = $scope.coParams[graphIndex];
       model.effect =  effectNames[graphIndex];
-      if ( hasSomeElements(model.coparams) ){
-        var message = 'Cost-coverage curve plotting options should be either empty, or all present!';
+      if ( !$scope.areValidParams(model.coparams) ){
         modalService.inform(
-          function (){ null },
+          function () {},
           'Okay',
-          message,
+          $scope.optionsErrorMessage,
           'Error!'
         );
         return;
       }
 
-      if (hasNoElements(model.ccparams)) delete model.ccparams;
-      if (hasNoElements(model.coparams)) delete model.coparams;
```

Comment:
Thanks, got it now. I didn't even think of the empty array case, because model.ccparams currently can't be an empty array. Because as soon as you initialise the UI you get an array filled with NaN or null depending on the input type.

I cleaned it up & made it explicit to make sure this will work even if we change things in the controller.
https://github.com/StarterSquad/Optima/commit/ef67c4e7bedbb157122a4d70fcd8c155063811aa

What do you think?